sql/scan.go

```diff
@@ -363,6 +371,66 @@ func (n *scanNode) initTargets(targets parser.SelectExprs) error {
 	return nil
 }
 
+var (
+	errLimitNoVariables  = errors.New("argument of LIMIT must not contain variables")
+	errOffsetNoVariables = errors.New("argument of OFFSET must not contain variables")
+)
+
+func (n *scanNode) initLimit(limitNode *parser.Limit) error {
+	if limitNode == nil {
+		n.limit, n.offset = parser.DNull, parser.DNull
+		return nil
+	}
+
+	if limitNode.Count == nil {
+		n.limit = parser.DNull
+	} else {
+		if parser.ContainsVars(limitNode.Count) {
+			n.err = errLimitNoVariables
+			return n.err
+		}
+
+		limitNode.Count, n.err = parser.NormalizeExpr(limitNode.Count)
```

Comment:
doesn't really matter since NormalizeExpr will go all the way and evaluate to a Datum in this case, so TypeCheck has nothing to add. perhaps NormalizeExpr shouldn't evaluate constants? I'm not sure.